src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -381,14 +382,32 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
       this._sourceObj = videoTracks[videoTrack.index];
       this._eventManager.listen(this._videoElement, Html5Events.LOADED_DATA, () => {
         this._eventManager.unlisten(this._videoElement, Html5Events.LOADED_DATA);
-        this._eventManager.listen(this._videoElement, Html5Events.SEEKED, () => {
-          this._eventManager.unlisten(this._videoElement, Html5Events.SEEKED);
-          this._onTrackChanged(videoTrack);
-        });
-        this._videoElement.currentTime = currentTime;
+        if (Env.browser.name === 'Android Browser') {
```

Comment:
@yairans can we make it use the actual data fro video and not browser sniffing based?
Is it possible to check after LOADED_DATA if video.duration exist and only if not then register DURATION_CHANGE and call play?
Also, if I recall this is a known bug that on Android stock browser the loadeddata event is fired before even the loadedmetadata is fired, can we move this implemntation to use loadedmetadata instead? if so then we need to test it again on all environments/platforms.